### ISSUE TITLE
ZeroDivisionError when plotting parallel_coordinate with matplotlib backend

### ISSUE THREAD

**Issue by appleparan** [optuna/optuna#2011]
## Expected behavior

Just plot parallel coordinate without Error

## Environment

- Optuna version: 2.3.0
- Python version: 3.8.2
- OS: Ubuntu 18.04
- (Optional) Other libraries and their versions:

## Error messages, stack traces, or logs

```
Traceback (most recent call last):
  File "mise/mise.py", line 106, in <module>
    ml(args)
  File "mise/mise.py", line 84, in ml
    globals()[f]()
  File "/home/appleparan/src/Mise.py/mise/ml/rnn_mul_tpa_attn_general.py", line 309, in ml_rnn_mul_tpa_attn_general
    ax_pcoord = optmpl.plot_parallel_coordinate(
  File "/home/appleparan/.pyenv/versions/mise/lib/python3.8/site-packages/optuna/_experimental.py", line 68, in new_func
    return func(*args, **kwargs)  # type: ignore
  File "/home/appleparan/.pyenv/versions/mise/lib/python3.8/site-packages/optuna/visualization/matplotlib/_parallel_coordinate.py", line 42, in plot_parallel_coordinate
    return _get_parallel_coordinate_plot(study, params)
  File "/home/appleparan/.pyenv/versions/mise/lib/python3.8/site-packages/optuna/visualization/matplotlib/_parallel_coordinate.py", line 96, in _get_parallel_coordinate_plot
    dims_obj_base[i].append((v - p_min) / p_w * obj_w + obj_min)
ZeroDivisionError: division by zero
```

## Steps to reproduce

1. Use Optuna in my code based on `pytorch_lightning_example.py` with pruning
2. ZeroDivisionError occurred sometimes. It is hard to provide MWE(Minimal Working Example) yet.

## Reproducible examples (optional)

```python
# python code
```

## Additional context (optional)
AFAIK, It seems the error occurred when parameter values has only single value. This makes `p_w` is zero throwing `ZeroDivisionError`.
https://github.com/optuna/optuna/blob/e54cc6f0d7864fb855aa97bfd0f30c5a28107562/optuna/visualization/matplotlib/_parallel_coordinate.py#L92-L96

Even I'm failed to provide MWE, I think there would be two solutions. 
1. Handle exception
2. Remove a possibility `p_w` to be zero. I think `parallel_coordinate` should be plotted even parameter `values` has only single value by lines just passing single point in figure.

**Comment by nzw0301** [comment]
I could reproduce your reported bug by running the following example:

```python
import optuna


def objective(trial):
    x = trial.suggest_uniform("x", 1, 2)
    y = trial.suggest_categorical("y", [0])
    return x ** 2 + y


sampler = optuna.samplers.TPESampler(seed=10)
study = optuna.create_study(sampler=sampler)
study.optimize(objective, n_trials=10)

optuna.visualization.matplotlib.plot_parallel_coordinate(study, params=["x", "y"])
```

Another example is 

```python
import optuna


def objective(trial):
    x = trial.suggest_uniform("x", 1, 2)
    y = trial.suggest_categorical("y", [0, 1])
    return x ** 2 + y


sampler = optuna.samplers.TPESampler(seed=10)
study = optuna.create_study(sampler=sampler)
study.optimize(objective, n_trials=1)  # if n_trials is small, then it might happen as well

optuna.visualization.matplotlib.plot_parallel_coordinate(study, params=["x", "y"])
```

**Comment by appleparan** [comment]
@nzw0301 Thanks for providing MWE (Minimum Working Example). As you commented, it happens when number of trials was small. 

In fact, I made a mistake on `timeout` when creating study, so I ran only few number of trials and got the error. I resolved this by using original visualization methods based on `Plotly`. However, in my opinion, this needs to be fixed.  

I wish I could made a PR, but still I don't understand how the scaling works and why needed in `matplotlib` only. If someone trying to fix this issue, I hope the reason of this exception , *only few trials*, helps you.

**Comment by nzw0301** [comment]
@ytknzw Could you explain the necessity of scaling that causes the error mentioned in this PR?

**Comment by nzw0301** [comment]
@appleparan I agree with you. `Plotly` backend does not have this error, so I also feel this error should be fixed.  As you comment, this error happens with high probability if the searched space is small and the number of trials is small.

**Comment by nzw0301** [comment]
```python
```


https://github.com/nzw0301/optuna/commit/1b2c694c91a2061d53b21c51ab92c44043ac7749

I'm not familiar with matplotlib, but

**Comment by nzw0301** [comment]
Hi @appleparan, your reported bug has been solved by #2354!

**Comment by crcrpar** [comment]
thank you @appleparan for reporting and @nzw0301 for the fix
